src/messages/messages.ts

```diff
@@ -99,3 +101,43 @@ export const createReactionMessage = (fromId: string, emoji: string, inReplyTo:
   fromId,
   inReplyTo,
 });
+
+const serialize = (message: DSNPMessage): string => {
+  const sortedMessage = sortObject((message as unknown) as Record<string, unknown>);
+  let serialization = "";
+
+  for (const key in sortedMessage) {
+    serialization = `${serialization}${key}${sortedMessage[key]}`;
+  }
+
+  return serialization;
+};
+
+/**
+ * sign() takes a DSNP message and returns a Uint8Array signature for the
+ * message using the Signer provided in the configuration options.
+ *
+ * @throws {@link MissingSigner}
+ * This error is thrown if no Signer is defined in the configuration options.
+ *
+ * @param message The DSNP message to sign
+ * @param opts    Optional. Configuration overrides, such as from address, if any
+ * @returns       The message signature in hex
+ */
+export const sign = async (message: DSNPMessage, opts?: Config): Promise<HexString> => {
+  const { signer } = await getConfig(opts);
+  if (!signer) throw MissingSigner;
```

Comment:
Side note, we should likely be throwing this in the config module instead. Perhaps `config.getSigner(opts)`, but I think that is a separate story.